conda/resolve.py

```diff
@@ -732,6 +733,10 @@ def get_dists(self, specs, max_only=False, filtered=False):
             self.clear_filter()
         return dists
 
+    def dependency_sort(self, dists):
+        sort_info = {name_dist(dist): dist for dist in dists}
```

Comment:
We shouldn't have a cross-dependency with `conda.install` here. Instead, we should robably pull the name directly from the Resolve object's index. That is, `self.index[dist]['name']`. Even if that fails, because the dist is not in the index, we can use `dist.rsplit('-',2)[0]`, or even better a new function to implement that.